File: Compare/Java_C#/GenericArray/Gen2.java
import java.util.*;

class Gen2 {
  public static void main(String[] args) {
    List<String>[] list = new ArrayList[1];
    Object[] ol = (Object[]) list;
    ol[0] = new ArrayList<Object>();//new List<Object>();
    ((List<Object>)(Object)list[0]).add(new Object());
    list[0].get(0).toString();
  }
}

File: Compare/Java_C#/GenericArray/Gen2.cs
using System.Collections.Generic;
using System;

class Gen2 {
  public static void Main(String[] args) {
    List<String>[] list = new List<String>[1];
    Object[] ol = (Object[]) list;
    ol[0] = new List<Object>();
    ol[0].ToString();
  }
}